src/utils/record-completeness/ComputeRecordCompleteness.ts

```diff
@@ -1,92 +1,113 @@
-import HTTPError from "../../models/HTTPError";
-import {ERRORS, RECORD_COMPLETENESS_ENUM, VEHICLE_TYPE} from "../../assets/Enums";
-import * as coreMandatoryValidation from "./CoreMandatoryValidations";
-import {ObjectSchema} from "@hapi/joi";
-import * as nonCoreMandatoryValidation from "./NonCoreMandatoryValidations";
-import { Vehicle, Trailer } from "../../../@Types/TechRecords";
-
-const validateRecordCompleteness = (validationSchema: ObjectSchema | undefined, techRecordFields: any) => {
-  if (validationSchema) {
-    return validationSchema.validate(techRecordFields, {stripUnknown: true});
-  } else {
-    return undefined;
+import { ValidationResult, ObjectSchema } from '@hapi/joi';
+import { Vehicle, Trailer } from '../../../@Types/TechRecords';
+import { ERRORS, RECORD_COMPLETENESS_ENUM, VEHICLE_TYPE } from '../../assets';
+import HTTPError from '../../models/HTTPError';
+import {
+  trlCoreMandatoryVehicleAttributes,
+  psvHgvCarLgvMotoCoreMandatoryVehicleAttributes,
+  carCoreMandatorySchema,
+  hgvCoreMandatorySchema,
+  lgvCoreMandatorySchema,
+  motorcycleCoreMandatorySchema,
+  psvCoreMandatorySchema,
+  trlCoreMandatorySchema
+} from './CoreMandatoryValidations';
+import { psvNonCoreMandatorySchema, hgvNonCoreMandatorySchema, trlNonCoreMandatorySchema } from './NonCoreMandatoryValidations';
+
+export function computeRecordCompleteness(vehicle: Vehicle): string {
+  if (!vehicle.systemNumber) {
+    throw new HTTPError(400, ERRORS.SYSTEM_NUMBER_GENERATION_FAILED);
   }
-};
 
-const validateVehicleAttributes = (vehicleType: string, vehicleAttributes: any) => {
-  if (vehicleType !== VEHICLE_TYPE.TRL) {
-    return validateRecordCompleteness(coreMandatoryValidation.psvHgvCarLgvMotoCoreMandatoryVehicleAttributes, vehicleAttributes);
+  const generalErrors = validateVehicleAttributes(vehicle)?.error;
+
+  if (generalErrors) {
+    console.log("general errors: ",generalErrors);
```

Comment:
Should this be displayed through VTM instead of a console log?

Reply:
Updating record completeness is a byproduct of creating and/or updating tech records. The result of its calculation does not trigger an error list. That can be implemented in the future if it's MVP and agreed upon by the design team.